### PR TITLE
build: add check for lib size

### PR DESCRIPTION
**Pull request by styfle**
This adds a check to make sure the lib size doesn't increase more than 10% after the "LKG" step.

This change was suggested [by mhegazy in this comment](https://github.com/Microsoft/TypeScript/pull/24282#issuecomment-391114630). 

Related #24282

### CODE REVIEW

File: scripts/build/get-dir-size.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+// @ts-check
+const { lstatSync, readdirSync } = require("fs");
+const { join } = require("path");
+const { promisify } = require("util");
+const execFile = promisify(require("child_process").execFile);
```
Comment (j-oliveras) [inline scripts/build/get-dir-size.js]: I don't know if is needed, but promisify not exists in node 6.

File: Gulpfile.js [hunk 2]
```diff
@@ -12,6 +12,7 @@ const clone = require("gulp-clone");
 const newer = require("gulp-newer");
 const tsc = require("gulp-typescript");
 const tsc_oop = require("./scripts/build/gulp-typescript-oop");
+const { getDirSize } = require("./scripts/build/get-dir-size");
```
Comment (mhegazy) [inline Gulpfile.js]: nit, do not use kebab-case, instead use camel case.

**Comment by mhegazy** [comment]
we also need to make a similar change to jakefile.js

File: scripts/build/get-dir-size.js (hunk 1 above)
Comment (DanielRosenwasser) [reply scripts/build/get-dir-size.js]: Also, `execFile` isn't used?

File: Gulpfile.js [hunk 3]
```diff
@@ -588,7 +589,14 @@ gulp.task("VerifyLKG", /*help*/ false, [], () => {
 gulp.task("LKGInternal", /*help*/ false, ["lib", "local"]);
 
 gulp.task("LKG", "Makes a new LKG out of the built js files", ["clean", "dontUseDebugMode"], () => {
-    return runSequence("LKGInternal", "VerifyLKG");
+    const lib = "./lib";
+    const sizeBefore = getDirSize(lib);
+    const seq = runSequence("LKGInternal", "VerifyLKG");
+    const sizeAfter = getDirSize(lib);
+    if (sizeAfter > (sizeBefore * 1.10)) {
+        throw new Error("The lib folder increased by 10% or more. This likely indicates a bug.");
```
Comment (DanielRosenwasser) [inline Gulpfile.js]: How do we want to confirm "no, this really is what we meant to do"?

Comment (mhegazy) [reply Gulpfile.js]: u can manually check in the LKG. this happens after the LKG has been built and updated.

File: scripts/build/get-dir-size.js (hunk 1 above)
Comment (styfle) [reply scripts/build/get-dir-size.js]: Oops, this was a bad copypasta on my part 😟

File: Gulpfile.js (hunk 2 above)
Comment (styfle) [reply Gulpfile.js]: Ok I changed it 🔧

**Comment by styfle** [comment]
@mhegazy Which line number in the `Jakefile.js`?
I'm guessing here line L659 but I'm not sure.

https://github.com/styfle/TypeScript/blob/7a5dec4d7b64743e5acc6d41c44d9907fe91250a/Jakefile.js#L659

**Comment by mhegazy** [comment]
I would do it after we have copied all expected files.

**Comment by styfle** [comment]
@mhegazy How does that look?

**Comment by mhegazy** [comment]
thanks!

**Pull request merged**